vault_service/views/user.py

```diff
@@ -195,4 +197,451 @@ def store_data():
+@bp.route('/get-myads/<user_id>', methods=['GET'])
+def get_myads(user_id):
+    '''
+    Fetches a myADS profile for the pipeline for a given uid
+    '''
+
+    # takes a given uid, grabs the user_data field, checks for the myADS key - if present, grabs the appropriate
+    # queries/data from the queries/myADS table. Finally, returns the appropriate info along w/ the rest
+    # of the setup from the dict (e.g. active, frequency, etc.)
+
+    # TODO check rate limit
+
+    # structure in vault:
+    # "myADS": [{"name": user-supplied name, "qid": QID from query table (type="query") or ID from myads table
+    # (type="template"), "type":  "query" or "template", "active": true/false, "frequency": "daily" or "weekly",
+    # "stateful": true/false}]
+
+    output = []
+    with current_app.session_scope() as session:
+        setups = session.query(MyADS).filter_by(user_id=user_id).filter_by(active=True).order_by(MyADS.id.asc()).all()
+        if not setups:
+            return '{}', 404
+        for s in setups:
+            o = {'id': s.id,
+                 'name': s.name,
+                 'type': s.type,
+                 'active': s.active,
+                 'stateful': s.stateful,
+                 'frequency': s.frequency,
+                 'template': s.template,
+                 'classes': s.classes,
+                 'data': s.data,
+                 'created': s.created.isoformat(),
+                 'updated': s.updated.isoformat()}
+
+            if s.type == 'query':
+                try:
+                    q = session.query(Query).filter_by(id=s.query_id).one()
+                    qid = q.qid
+                except ormexc.NoResultFound:
+                    qid = None
+            else:
+                qid = None
+            o['qid'] = qid
+
+            output.append(o)
+
+    return json.dumps(output), 200
+
+
+@advertise(scopes=['ads-consumer:myads'], rate_limit = [1000, 3600*24])
+@bp.route('/myads-users/<iso_datestring>', methods=['GET'])
+def export(iso_datestring):
+    '''
+    Get the latest changes (as recorded in users table)
+        The optional argument latest_point is RFC3339, ie. '2008-09-03T20:56:35.450686Z'
+    '''
+    # TODO check rate limit
+    # TODO need to add a created/update field to users table
+
+    # inspired by orcid-service endpoint of same endpoint, checks the users table for profiles w/ a myADS setup that
+    # have been updated since a given date/time; returns these profiles to be added to the myADS processing
+
+    latest = parser.parse(iso_datestring) # ISO 8601
+    output = []
+    with current_app.session_scope() as session:
+        setups = session.query(MyADS).filter(MyADS.updated > latest).order_by(MyADS.updated.asc()).all()
+        for s in setups:
+            output.append(s.user_id)
+
+    output = list(set(output))
+    return json.dumps({'users': output}), 200
+
+
+@advertise(scopes=['import-myads'], rate_limit=[1000, 3600*24])
```

Comment:
This is also a new scope, so when deployed, none of our users will be able to use these endpoints. We would need to update the database (for all non-anonymous users) and change adsws config: https://github.com/adsabs/BeeHive/blob/00730df6d0582db59b7932a1567393c46df2bccf/images/common/adsws/local_config.py#L97